pkg/oauthserver/help_test.go

```diff
@@ -0,0 +1,97 @@
+package oauthserver_test
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/openshift/oauth-server/pkg/audit"
+	apiaudit "k8s.io/apiserver/pkg/apis/audit"
+	kaudit "k8s.io/apiserver/pkg/audit"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+)
+
+func withKeyValues(values ...[2]string) string {
+	const (
+		key   = 0
+		value = 1
+	)
+
+	form := url.Values{}
+
+	for _, tuple := range values {
+		form.Add(tuple[key], tuple[value])
+	}
+
+	return form.Encode()
+}
+
+func withKeyValue(key, value string) [2]string {
+	return [2]string{key, value}
+}
+
+func WithAuth(req *http.Request, username, password string) {
+	req.Header.Add(
+		"Authorization",
+		"Basic "+base64.StdEncoding.EncodeToString([]byte(username+":"+password)),
+	)
+}
+
+func withDecisionAnnotation(want string) func(map[string]string) error {
```

Comment:
no error returning test check functions, pass `*testing.T` instead

Reply:
Is this idiomatic? I prefer to return all the errors to the executing func and let it handle there.

Reply:
I prefer pure functions that have no side effects, but I will do as you suggest.